### DISCUSSION TITLE
Issues with Media cache retention [URGENT]

### DISCUSSION THREAD

**Discussion by compi2k** [mastodon/mastodon#30373] [Q&A]
Hello, I am having issues with media cache retention. I have it set to 7 days.

Admin panel reports 21.1GB being used:
![image](https://github.com/mastodon/mastodon/assets/65253326/1620b6b2-7a66-408c-9293-b539f160da4e)

On the other hand, Backblaze Reports a whopping 295GB and 1.4 million files:
![image](https://github.com/mastodon/mastodon/assets/65253326/22d723b2-62e2-45ba-a7df-8aa7f5691f23)

Most is in cache dir:
![image](https://github.com/mastodon/mastodon/assets/65253326/3d1b372c-cc63-4640-af3c-5dafe45b0f4e)

Something is clearly wrong and not being deleted correctly. I tried to remove orphaned files with tootctl, but that only finds roughly 300MB: 
![image](https://github.com/mastodon/mastodon/assets/65253326/88d26146-f8b5-4cf6-8d88-728bdecf538c)

Media remove also only gets rid of 5GB, still far away from 21.1GB:
![image](https://github.com/mastodon/mastodon/assets/65253326/a51f3d68-b4ec-4680-9eea-20ffb02d4cb2)

Please help me get rid of this 200GB

**Comment by compi2k** [comment]
push, still happening

**Comment by StrangeMinds1960** [comment]
Have you tried tootctl cache clear?

https://docs.joinmastodon.org/admin/tootctl/#cache-clear

(Use at your own risk)

**Comment by quicoto** [comment]
You might have better luck posting on #MastoAdmin hashtag directly rather than here, seeing the urgency of the topic

I wrote a blog post with what I use to keep my media object storage lean, maybe you can get inspired: https://ricard.dev/improving-mastodons-disk-usage/
